### DISCUSSION TITLE
Issues with joint limits for continuous, revolute joints

### DISCUSSION THREAD

**Discussion by varshiths** [bulletphysics/bullet3#3780] [General]
ROS makes a distinction between `revolute` and `continuous` joints as those with and without joint limits, while it would seem that bullet does not.
It so happened that in the URDF file I was using, the joint was marked `continuous`, but the `lower` and `upper` limits were specified to be 0, 0. While it was certainly redundant, it resulted in the joint always being restricted to 0 in the bullet simulation.

I am not sure of the implementation, but it seems that joints marked `continuous` and `revolute` are treated the exact same way, with limits `0`, `-1` being added if not specified, irrespective of the type of the joint.

While not a major issue, this is clearly wrong.
The limits should be overwritten in the case of a `continuous` joint.

**Comment by erwincoumans** [comment]
It is a bit odd to specifiy joint limits while also using the 'continuous' type, better make up your mind if you want limits or not.
Indeed, we should protect the user from this indecisiveness.
